### PR TITLE
Boomer glutton and monstergroup additions, added spell and emit field

### PR DESCRIPTION
**Pull request by SenpaiSlime**
#### Summary
Content "Added the new boomer glutton and boomer evolution group."

#### Purpose of change
To give the players a new terrifying close range threat, which requires a more unique approach especially indoors, and to allow boomers the evolution tree they deserve, instead of the single choice they had before. Also adds in a new sludge field spell, and a sludge emit field, both for the glutton.

#### Describe the solution
Added in the new, freaky boomer glutton, and his special emit and attack. Gave the boomer its own monstergroup to evolve into.

#### Testing
Have done a bit of spawning in and fighting the creature to make sure it does everything it is supposed to, sent a copy of the files to another user to also test out, and i'm very pleased with the results from what we've seen.

### CODE REVIEW

File: data/json/emit.json [hunk 1]
```diff
@@ -360,5 +360,14 @@
     "field": "fd_nuke_gas",
     "intensity": 3,
     "qty": 100
+  },
+  {
+    "id": "emit_sludge_puddle",
+    "type": "emit",
+    "//": "infrequent emission of sludge along the ground, made because sludge trail is too frequent",
+    "field": "fd_sludge",
+    "intensity": 3,
```
Comment (casswedson) [inline data/json/emit.json:369]: This file still needs linting.
image

File: data/json/monster_special_attacks/spells.json [hunk 2]
```diff
@@ -294,5 +294,27 @@
     "effect": "summon",
     "effect_str": "mon_dermatik_larva",
     "extra_effects": [ { "id": "frog_mother_hurt_self", "hit_self": true } ]
+  },
+  {
+    "type": "SPELL",
+    "id": "spread_sludge",
+    "name": { "str": "Spread Sludge" },
+    "description": "sludge splatters along the ground.",
```
Comment (casswedson) [inline data/json/monster_special_attacks/spells.json:302]:
```suggestion
    "description": "Sludge splatters along the ground.",
```

File: data/json/monsters/zed_explosive.json [hunk 3]
```diff
@@ -163,5 +163,69 @@
     "death_drops": "default_zombie_items",
     "death_function": [ "GAS" ],
     "flags": [ "SEES", "HEARS", "SMELLS", "STUMBLES", "WARM", "POISON", "NO_BREATHE", "REVIVES", "FILTHY" ]
+  },
+  {
+    "id": "mon_boomer_glutton",
+    "type": "MONSTER",
+    "name": "boomer glutton",
+    "description": "A grotesque ball of flesh and fat. Seemingly vestigial limbs flop around haphazardly as it rolls across the ground. You occasionally catch a glimpse of bulging eyes and a distended gaping maw oozing some sort of black sludge.",
```
Comment (casswedson) [inline data/json/monsters/zed_explosive.json:171]:
```suggestion
    "description": "A grotesque ball of flesh and fat.  Seemingly vestigial limbs flop around haphazardly as it rolls across the ground.  You occasionally catch a glimpse of bulging eyes and a distended gaping maw oozing some sort of black sludge.",
```

File: data/json/monsters/zed_explosive.json [hunk 4]
```diff
@@ -163,5 +163,69 @@
...
+        "no_infection_chance": 3,
+        "min_mul": 0.75,
+        "max_mul": 1.25,
+        "damage_max_instance": [ { "damage_type": "stab", "amount": 10, "armor_multiplier": 0.5 } ]
+      },
+      {
+        "type": "spell",
+        "spell_data": { "id": "spread_sludge", "min_level": 1 },
+        "cooldown": 28,
+        "monster_message": "the glutton vomits sludge!"
```
Comment (casswedson) [inline data/json/monsters/zed_explosive.json:209]:
```suggestion
        "monster_message": "The glutton vomits sludge!"
```

File: data/json/emit.json (hunk 1 above)
Comment (SenpaiSlime) [reply data/json/emit.json:369]: ill fix that ASAP

File: data/json/monsters/zed_explosive.json [hunk 5]
```diff
@@ -171,5 +171,63 @@
...
+      },
+      {
+        "type": "spell",
+        "spell_data": { "id": "spread_sludge", "min_level": 1 },
+        "cooldown": 28,
+        "monster_message": "The glutton vomits sludge!"
+      }
+    ],
+    "death_drops": "default_zombie_items",
+    "death_function": [ "BOOMER" ],
```
Comment (kevingranade) [inline data/json/monsters/zed_explosive.json:217]:
```suggestion
     "death_function": { "effect": { "id": "death_boomer", "hit_self": true }, "message": "A %s explode!", "corpse_type": "NO_CORPSE" },
```

**Comment by SenpaiSlime** [comment]
If anyone wants to drop this into their game and try it out, i would appreciate the feedback!

**Pull request merged**